### PR TITLE
OADP-5106: Add legacy-aws default plugin reconcile e2e test.

### PR DESCRIPTION
**Pull request by kaovilai**
- **Add legacy-aws default plugin reconcile e2e test.**

## Why the changes were made

To check that reconcile against legacy-aws works.

This will go in after #1565

### CODE REVIEW

File: tests/e2e/dpa_deployment_suite_test.go [hunk 1]
```diff
@@ -344,6 +348,12 @@ var _ = ginkgo.Describe("Configuration testing for DPA Custom Resource", func()
 				DoNotBackupImages:       true,
 			}),
 		}),
+		ginkgo.Entry("DPA CR with legacy-aws plugin", ginkgo.Label("aws"), InstallCase{
```
[the remark this answers is not in the document]
Comment (kaovilai) [reply tests/e2e/dpa_deployment_suite_test.go:351]: Sure

[the remark this answers is not in the document]
Comment (kaovilai) [reply tests/e2e/dpa_deployment_suite_test.go]: resolved.

[the remark this answers is not in the document]
Comment (kaovilai) [reply tests/e2e/dpa_deployment_suite_test.go]: https://github.com/openshift/oadp-operator/compare/cecea01ca905b4181b0e672df2997544a874c79d..9f4f221b3ca62597a44a3c1762aee4d8925fc9e9

**Review by mateusoliveira43 — APPROVED** [review]
This fixes related issue, but I would also suggest duplicating all unit/integration tests for aws plugin for legacy-aws plugin

**Comment by sseago** [comment]
@kaovilai It might be good to also add a test that adds both aws and legacy-aws and verifies that this produces a reconcile error.

**Comment by weshayutin** [comment]
```
Configuration testing for DPA Custom Resource
/go/src/github.com/openshift/oadp-operator/tests/e2e/dpa_deployment_suite_test.go:134
  DPA reconciled to true
  /go/src/github.com/openshift/oadp-operator/tests/e2e/dpa_deployment_suite_test.go:145
    DPA CR with legacy-aws plugin [aws, ibmcloud]
    /go/src/github.com/openshift/oadp-operator/tests/e2e/dpa_deployment_suite_test.go:351
  > Enter [It] DPA CR with legacy-aws plugin - /go/src/github.com/openshift/oadp-operator/tests/e2e/dpa_deployment_suite_test.go:351 @ 10/24/24 08:00:50.211
2024/10/24 08:00:50 DPA status is Reconciled: True, reason Complete: Reconcile complete
2024/10/24 08:00:50 DPA status is Reconciled: True, reason Complete: Reconcile complete
2024/10/24 08:01:05 DPA status is Reconciled: True, reason Complete: Reconcile complete
2024/10/24 08:01:20 DPA status is Reconciled: True, reason Complete: Reconcile complete
2024/10/24 08:01:35 DPA status is Reconciled: True, reason Complete: Reconcile complete
2024/10/24 08:01:50 Waiting for velero Pod to be running
2024/10/24 08:01:50 velero Pod phase is Running
2024/10/24 08:02:05 velero Pod phase is Running
2024/10/24 08:02:20 velero Pod phase is Running
2024/10/24 08:02:35 velero Pod phase is Running
2024/10/24 08:02:50 Waiting for NodeAgent DaemonSet to be deleted
2024/10/24 08:02:50 Checking if NodeAgent DaemonSet has been deleted...
2024/10/24 08:02:50 Checking if BSLs are available
2024/10/24 08:03:10 BSL ts-velero-test-1 phase is Available
2024/10/24 08:03:10 Checking for BSL spec
2024/10/24 08:03:10 Checking no VSLs are deployed
2024/10/24 08:03:10 Waiting for velero Deployment to have expected default plugins
2024/10/24 08:03:10 Checking for openshift default plugin
2024/10/24 08:03:10 Checking for legacy-aws default plugin
2024/10/24 08:03:10 Waiting for 1 minute after reconciled: 1m20.093673798s elapsed
```

**Review by weshayutin — APPROVED** [review]
/LGTM

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**